recipe/bld.bat

```diff
@@ -0,0 +1,66 @@
+echo on
+set "PYPY3_SRC_DIR=%SRC_DIR%\pypy3"
+set "PYTHON=%SRC_DIR%\pypy2-binary\pypy.exe"
+
+REM PyPy translation looks for this.
+set "PYPY_LOCALBASE=%PREFIX%"
+
+set "GOAL_DIR=%PYPY3_SRC_DIR%\pypy\goal"
+set "RELEASE_DIR=%PYPY3_SRC_DIR%\pypy\tool\release"
+
+set "PYPY_PKG_NAME=pypy3"
+set "BUILD_DIR=%PREFIX%\..\build"
+set "TARGET_DIR=%PREFIX%\..\target"
+set "ARCHIVE_NAME=%PYPY_PKG_NAME%-%PKG_VERSION%"
+
+REM Build PyPy.
+cd /d %GOAL_DIR%
+REM This is what we would like to do
+rem %PYTHON% ..\..\rpython\bin\rpython --make-jobs %CPU_COUNT% --shared -Ojit targetpypystandalone.py
+rem But the machine runs out of memory, so break it into parts
+rem -----------------
+set "PYPY_USESSION_BASENAME=pypy3"
+%PYTHON% ..\..\rpython\bin\rpython --no-compile --shared -Ojit targetpypystandalone.py
+cd /d %TEMP%\usession-pypy3-0\testing_1 || exit /b 11
+dir Makefile || exit /b 11
+jom /f Makefile || exit /b 11
+copy *.pdb %GOAL_DIR% 
+copy *.dll %GOAL_DIR% 
+copy *.exe %GOAL_DIR% 
+rem TODO: parameterize this
+copy libpypy3-c.lib %PYPY3_SRC_DIR%\libs\python37.lib || exit /b 11
+cd /d %GOAL_DIR%
+rem -----------------
+
+REM Build cffi imports using the generated PyPy.
+set PYTHONPATH=..\..
+%PYPY_PKG_NAME%-c ..\..\lib_pypy\pypy_tools\build_cffi_imports.py
+set PYTHONPATH=
+
+REM Package PyPy.
+mkdir %TARGET_DIR%
+
+%PYTHON% %RELEASE_DIR%\package.py --builddir="%TARGETDIR% --targetdir="%TARGET_DIR%" --archive-name="%ARCHIVE_NAME%"
+
+REM Move all files from the package to conda's $PREFIX.
+robocopy /S %TARGET_DIR%\%ARCHIVE_NAME% %PREFIX%
+
+REM Move the generic file name to somewhere that's specific to pypy
+move %PREFIX%\README.rst %PREFIX%\lib_pypy\
+REM License is packaged separately
+del %PREFIX%\LICENSE
+
+REM Make sure the site-packages dir match with cpython
+set PY_VERSION=%name_suffix%
+mkdir  %PREFIX%\lib\python%PY_VERSION%\site-packages
+move %PREFIX%\site-packages\README %PREFIX%\lib\python%PY_VERSION%\site-packages\
+rmdir /q /s %PREFIX%\site-packages
+mklink /D %PREFIX%\site-packages %PREFIX%\lib\python%PY_VERSION%\site-packages
```

Comment:
`mklink` doesn't work without administrator privileges and conda-build doesn't support symlinks on windows AFAIK.
What we need is a way to tell PyPy that the site-packages need to be `%PREFIX%\Lib\site-packages` on windows. Relevant code https://github.com/mozillazg/pypy/blob/release-pypy3.7-v7.3.5/lib-python/3/site.py#L347-L356

Reply:
Something like,
```
diff --git a/lib-python/3/site.py b/lib-python/3/site.py
index 84de242aee..51d7a66fa3 100644
--- a/lib-python/3/site.py
+++ b/lib-python/3/site.py
@@ -344,9 +344,6 @@ def getsitepackages(prefixes=None):
             continue
         seen.add(prefix)
 
-        if is_pypy:
-            sitepackages.append(os.path.join(prefix, "site-packages"))
-            continue
         if os.sep == '/':
             sitepackages.append(os.path.join(prefix, "lib",
                                         "python%d.%d" % sys.version_info[:2],
diff --git a/lib-python/3/sysconfig.py b/lib-python/3/sysconfig.py
index d9eb7bfe79..884c7b04d5 100644
--- a/lib-python/3/sysconfig.py
+++ b/lib-python/3/sysconfig.py
@@ -44,8 +44,8 @@ _INSTALL_SCHEMES = {
     'pypy': {
         'stdlib': '{installed_base}/lib-{implementation_lower}',
         'platstdlib': '{base}/lib-{implementation_lower}',
-        'purelib': '{base}/site-packages',
-        'platlib': '{base}/site-packages',
+        'purelib': '{base}/lib/python{py_version_short}/site-packages',
+        'platlib': '{platbase}/lib/python{py_version_short}/site-packages',
         'include': '{installed_base}/include',
         'platinclude': '{installed_base}/include',
         'scripts': '{base}/bin',
@@ -64,8 +64,8 @@ _INSTALL_SCHEMES = {
     'pypy_nt': {
         'stdlib': '{installed_base}/lib-{implementation_lower}',
         'platstdlib': '{base}/lib-{implementation_lower}',
-        'purelib': '{base}/site-packages',
-        'platlib': '{base}/site-packages',
+        'purelib': '{base}/Lib/site-packages',
+        'platlib': '{base}/Lib/site-packages',
         'include': '{installed_base}/include',
         'platinclude': '{installed_base}/include',
         'scripts': '{base}/Scripts',
```

Reply:
thanks. Should we remove the symlink on linux as well?

Reply:
I don't think we can do that now. We can do it for pypy3.8 or the next version that breaks ABI compat.

Reply:
What do you think about using `pypy` in the {base}/lib/pypy{py_version_short}/site-packages like I proposed in https://github.com/python/cpython/pull/24628 (which has not been accepted yet)?

Reply:
The issue is that conda-build knows only about `{base}/lib/python{py_version_short}/site-packages` (unix) and `{base}/Lib/site-packages` (win)  and `noarch: python` packages are extracted into that folder. https://github.com/conda/conda/blob/33a142c16530fcdada6c377486f1c1a385738a96/conda/common/path.py#L171

Reply:
hmm. That should be using one of the sysconfig templates, but changing it now will probably break the world. I will go with a patch for now to revert most of site.py and sysconfig.py to cpython's version like you suggest.